controllers/flinkcluster_reconciler.go

```diff
@@ -140,9 +140,16 @@ func (reconciler *ClusterReconciler) reconcileDeployment(
 	if desiredDeployment != nil && observedDeployment != nil {
 		if getUpdateState(reconciler.observed) == UpdateStateUpdating {
 			updateComponent := fmt.Sprintf("%v deployment", component)
-			err := reconciler.deleteOldComponent(desiredDeployment, observedDeployment, updateComponent)
-			if err != nil {
-				return err
+			if *reconciler.observed.cluster.Spec.RecreateComponent {
+				err := reconciler.deleteOldComponent(desiredDeployment, observedDeployment, updateComponent)
+				if err != nil {
+					return err
+				}
+			} else {
+				err := reconciler.updateDeployment(desiredDeployment, updateComponent)
```

Comment:
What do you think about using one `updateComponent` function rather than each `updateXXX` function like `deleteOldComponent`? Is there anything better about using each `updateXXX` function?

Reply:
Frankly speaking，updateComponent is much better.